File: Frontend/src/app/settings/settings.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { environment } from 'src/environments/environment';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class SettingsService {
  APIURL = environment.BASE_URL;

  constructor(private http: HttpClient) { }

  UpdateAdminAccount(data: any) {
    return this.http.put(this.APIURL + '/Settings/updateadminaccount', data, { responseType: 'json' });
  }

  GetAdminUser(RoleId: any) {
    return this.http.get(this.APIURL + '/Settings/getbyroleId/' + RoleId, { responseType: 'json' });
  }

  CreateAdminUser(data: any): Observable<any> {
    return this.http.post(this.APIURL + '/Settings/saveadminuser', data, { responseType: 'json' });
  }

  UpdateAdminUser(data: any) {
    return this.http.put(this.APIURL + '/Settings/updateadminuser', data, { responseType: 'json' });
  }

  DeleteAdminUser(data: any) {
    return this.http.put(this.APIURL + '/Settings/deleteadminuser', data, { responseType: 'json' });
  }

  AdminUserAccountStatus(data: any) {
    return this.http.put(this.APIURL + '/Settings/adminuseraccountstatus', data, { responseType: 'json' });
  }

  GetRoleFeatureMapping(roleId: any) {
    return this.http.get(this.APIURL + '/Settings/getrolefeaturemapping/' + roleId, { responseType: 'json' });
  }

  GetUserFeatureMapping(UserId: any) {
    return this.http.get(this.APIURL + '/Settings/getuserfeaturemapping/' + UserId, { responseType: 'json' });
  }

  UpdateOwnerAccount(data : any){
    return this.http.put(this.APIURL + '/Settings/updateowneraccount',data,{responseType : 'json'});
  }

  GetEmailTemplate(){
    return this.http.get(this.APIURL + '/Settings/getemailtemplate',{responseType : 'json'});
  }

  UpdateEmailTemplate(data : any){
    return this.http.put(this.APIURL + '/Settings/updateemailtemplate',data,{responseType : 'json'});
  }

  UpdateEmailTemplateStatus(data : any){
    return this.http.put(this.APIURL + '/Settings/activeinactiveemailtemplate',data,{responseType : 'json'});
  }

  GetCredorInfo(){
    return this.http.get(this.APIURL + '/Settings/credorinfo',{responseType : 'json'});
  }

  UpdateCredorInfo(data : any){
    return this.http.put(this.APIURL + '/Settings/credorinfoupdate',data ,{responseType : 'json'});
  }
}

File: Backend/Credor.Web.API/Services/SettingsService.cs
using Credor.Client.Entities;
using Credor.Web.API.Common.UnitOfWork;
using Microsoft.Extensions.Configuration;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Credor.Web.API
{
    public class SettingsService : ISettingsService
    {
        private readonly UnitOfWork _unitOfWork;
        private readonly ISettingsRepository _settingsRepository;

        public IConfiguration _configuration { get; }

        public SettingsService(ISettingsRepository settingsRepository,IConfiguration configuration)
        {
            _settingsRepository = settingsRepository;
            _configuration = configuration;
            _unitOfWork = new UnitOfWork(configuration);
        }

        public List<AdminAccountDto> GetByRoleId(int RoleId)
        {
            return _settingsRepository.GetByRoleId(RoleId);
        }

        public int SaveAdminUser(AdminAccountDto adminAccountDto)
        {
            return _settingsRepository.SaveAdminUser(adminAccountDto);
        }

        public int UpdateAdminUser(AdminAccountDto adminAccountDto)
        {
            return _settingsRepository.UpdateAdminUser(adminAccountDto);
        }

        public int DeleteAdminUser(AdminAccountDto adminAccountDto)
        {
            return _settingsRepository.DeleteAdminUser(adminAccountDto);
        }

        public Task<int> UpdateAdminAccount(UpdateAdminAccountDto updateAdminAccountDto)
        {
            return _settingsRepository.UpdateAdminAccount(updateAdminAccountDto);
        }

        public int AdminUserAccountStatus(AdminAccountDto adminAccountDto)
        {
            return _settingsRepository.AdminUserAccountStatus(adminAccountDto);
        }

        public List<RoleFeatureMappingDto> GetRoleFeatureMapping(int RoleId)
        {
            return _settingsRepository.GetRoleFeatureMapping(RoleId);
        }

        public List<UserFeatureMappingDto> GetUserFeatureMapping(int userid)
        {
            return _settingsRepository.GetUserFeatureMapping(userid);
        }

        public int UpdateOwnerAccount(AdminAccountDto adminAccountDto)
        {
            return _settingsRepository.UpdateOwnerAccount(adminAccountDto);
        }

        public List<CredorEmailTemplateDto> GetEmailTemplate()
        {
            return _settingsRepository.GetEmailTemplate();
        }

        public int UpdateEmailTemplate(CredorEmailTemplateDto credorEmailTemplate)
        {
            return _settingsRepository.UpdateEmailTemplate(credorEmailTemplate);
        }

        public int ActiveInactiveEmailTemplate(CredorEmailTemplateDto credorEmailTemplate)
        {
            return _settingsRepository.ActiveInactiveEmailTemplate(credorEmailTemplate);
        }

        public List<CredorInfoDto> CredorInfo()
        {
            return _settingsRepository.CredorInfo();
        }

        public int CredorInfoUpdate(CredorInfoDto credorInfoDto)
        {
            return _settingsRepository.CredorInfoUpdate(credorInfoDto);
        }
    }
}
